### PR TITLE
added redirection function

### LINKED ISSUE #801: Livepeer.com Verify Page is not redirect correctly

### PR DESCRIPTION
**Pull request by FullStackTiger**
**What does this pull request do? Explain your changes. (required)**

**Specific updates (required)**

 - added redirection function

**Does this pull request close any open issues?**

- Fixes #801 

**Screenshots (optional):**

**Checklist:**

- [ ] I have read the **CONTRIBUTING** document.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [ ] I have added tests to cover my changes.

### CODE REVIEW

File: packages/www/pages/verify.jsx [hunk 1]
```diff
@@ -14,9 +14,28 @@ import Guides from "components/Marketing/Guides";
 const Verify = () => {
   useLoggedIn();
   const router = useRouter();
-  const { user } = useApi();
+  const { user, verify } = useApi();
   const { email, emailValidToken } = router.query;
 
+  useEffect(() => {
+    if (email && emailValidToken) {
+      verify(email, emailValidToken).then(() => {
```
Comment (victorges) [inline packages/www/pages/verify.jsx]: Does `verify` throw an exception in case verification fails? (e.g. invalid token)
What would happen on that case? Any easy way to display an error, maybe with a toast or sth?

No problem if there's currently no pattern for that in the dashboard and it's harder than it should be (I know we have a current tech debt for not showing errors to the user on most ops).

File: packages/www/pages/verify.jsx [hunk 2]
```diff
@@ -14,9 +14,28 @@ import Guides from "components/Marketing/Guides";
...
+        } else {
+          router.replace("/dashboard");
+        }
+      });
+    }
+  }, [email, emailValidToken]);
+
+  // If they've already validated their email, get 'em out of here
+  useEffect(() => {
+    if (user && user.emailValid !== false) {
```
Comment (victorges) [inline packages/www/pages/verify.jsx]: **nit:**
```suggestion
    if (user?.emailValid === true) {
```
since `user.emailValid: undefined` would also pass the current test and we probably don't want undefined/null (or anything not a boolean) to be considered a validated user.

File: packages/www/pages/verify.jsx [hunk 3]
```diff
@@ -14,9 +14,28 @@ import Guides from "components/Marketing/Guides";
 const Verify = () => {
   useLoggedIn();
   const router = useRouter();
-  const { user } = useApi();
+  const { user, verify } = useApi();
   const { email, emailValidToken } = router.query;
 
+  useEffect(() => {
```
Comment (victorges) [inline packages/www/pages/verify.jsx:20]: So, in practice, we weren't validating the user's email anymore even when they did click on the link received on the email? Good thing we got this back.

**Review by victorges — APPROVED** [review]
LGTM!

File: packages/www/pages/verify.jsx (hunk 3 above)
Comment (FullStackTiger) [reply packages/www/pages/verify.jsx:20]: if they did click on the link received on the email, we will have the user's email validating, so emailValid will be true.

File: packages/www/pages/verify.jsx (hunk 1 above)
Comment (FullStackTiger) [reply packages/www/pages/verify.jsx]: in case verification fails, does the user need to signup again?

Comment (FullStackTiger) [reply packages/www/pages/verify.jsx]: so we need to display the text like "We got an invalid token, you need to [signup](https://livepeer.com/) again"?

Comment (FullStackTiger) [reply packages/www/pages/verify.jsx]: in case verification fails, added the text like "Email verification failed: ${err}. Please access the link you received by email again."
